src/effects/lv2/lv2effectprocessor.cpp

```diff
@@ -0,0 +1,102 @@
+#include "effects/lv2/lv2effectprocessor.h"
+#include "engine/effects/engineeffect.h"
+#include "control/controlobject.h"
+#include "util/sample.h"
+
+#define MAX_BUFFER_LEN 160000
+
+LV2EffectProcessor::LV2EffectProcessor(EngineEffect* pEngineEffect,
+                                       EffectManifestPointer pManifest,
+                                       const LilvPlugin* plugin,
+                                       QList<int> audioPortIndices,
+                                       QList<int> controlPortIndices) {
+    m_sampleRate = ControlObject::get(ConfigKey("[Master]", "samplerate"));
+ 
+    m_handle = lilv_plugin_instantiate(plugin, m_sampleRate, NULL); 
```

Comment:
A separate instance of the plugin for each potential input channel is necessary. When multiple inputs are routed to an effect unit with an LV2 plugin loaded, the state is shared between the two channels which produces distorted sounds.